mlrun/model_monitoring/helpers.py

```diff
@@ -35,31 +36,21 @@ def get_stream_path(project: str = None):
     )
 
 
-def get_connection_string(project: str = None):
-    """Get endpoint store connection string from the project secret.
-    If wasn't set, take it from the system configurations"""
+def get_connection_string(secret_provider: typing.Callable = None) -> str:
+    """Get endpoint store connection string from the project secret. If wasn't set, take it from the system
+    configurations.
 
-    if is_running_as_api():
-        # Running on API server side
-        import mlrun.api.crud.secrets
-        import mlrun.common.schemas
+    :param secret_provider: An optional secret provider which in this case is a callable function that handles the
+                            connection string secret in the API side.
```

Comment:
I don't really see why the implementation details relevant here
Is it not possible that one day we'll want to use secret store or something else? (same for all other places you added this doc)
```suggestion
    :param secret_provider: An optional secret provider to get the connection string secret.
```